relay-server/src/services/spooler/envelope_stack/sqlite.rs

```diff
@@ -90,41 +90,47 @@ impl SQLiteEnvelopeStack {
     /// In case there is a failure while writing envelopes, all the envelopes that were enqueued
     /// to be written to disk are lost. The explanation for this behavior can be found in the body
     /// of the method.
-    async fn spool_to_disk(&mut self) -> Result<(), SQLiteEnvelopeStackError> {
+    fn background_spool_to_disk(&mut self) {
         let Some(envelopes) = self.batches_buffer.pop_front() else {
-            return Ok(());
+            return;
         };
+        // We optimistically assume that such elements were removed from the buffer, so we update
+        // the new size.
         self.batches_buffer_size -= envelopes.len();
 
-        let insert_envelopes = envelopes.iter().map(|e| InsertEnvelope {
-            received_at: received_at(e),
-            own_key: self.own_key,
-            sampling_key: self.sampling_key,
-            encoded_envelope: e.to_vec().unwrap(),
+        let insert_envelopes = envelopes
+            .iter()
+            .map(|e| InsertEnvelope {
+                received_at: received_at(e),
+                own_key: self.own_key,
+                sampling_key: self.sampling_key,
+                encoded_envelope: e.to_vec().unwrap(),
+            })
+            .collect();
```

Comment:
nit: I would move `envelopes` to the background task unaltered, and take from it in the `push_values` closure -> one vector less.